packages/rest/src/rest.server.ts

```diff
@@ -703,6 +713,9 @@ export class RestServer extends Context implements Server, HttpServerLike {
     let spec = this.getSync<OpenApiSpec>(RestBindings.API_SPEC);
     const defs = this.httpHandler.getApiDefinitions();
 
+    // Apply shallow-clone to prevent modification of user-provided API_SPEC
```

Comment:
@bajtos could you explain more about the shallow clone here?
Wouldn't it ALLOW modification of user-provided API_SPEC?

The line below it(L#706) applies deep clone which prevents the modification of the original spec.

Reply:
This is just a temporary workaround to get my demo working. The actual implementation is likely to be different, see the discussion in _Dynamic binding/rebinding of controllers after app start_ #433.